WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Service/BlogDashboardService.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+import WordPressKit
+
+class BlogDashboardService {
+    let remoteService: DashboardServiceRemote
+
+    init(managedObjectContext: NSManagedObjectContext, remoteService: DashboardServiceRemote? = nil) {
+        self.remoteService = remoteService ?? DashboardServiceRemote(wordPressComRestApi: WordPressComRestApi.defaultApi(in: managedObjectContext, localeKey: WordPressComRestApi.LocaleKeyV2))
+    }
+
+    func fetch(wpComID: Int, completion: @escaping (DashboardSnapshot) -> Void) {
+        let cardsToFetch: [String] = DashboardCard.allCases
+            .filter { $0.isRemote }
+            .map { $0.rawValue }
+
+        remoteService.fetch(cards: cardsToFetch, forBlogID: wpComID, success: { [weak self] cards in
+
+            var snapshot = DashboardSnapshot()
+
+            DashboardCard.allCases.forEach { card in
+
+                if card.isRemote {
+
+                    if card == .posts,
+                       let posts = cards[DashboardCard.posts.rawValue] as? NSDictionary,
+                       let (sections, items) = self?.parsePostCard(posts) {
+                        snapshot.appendSections(sections)
+                        sections.enumerated().forEach { key, section in
+                            snapshot.appendItems([items[key]], toSection: section)
+                        }
+                    } else {
+
+                        if let viewModel = cards[card.rawValue] {
+                            let section = DashboardCardSection(id: card.rawValue)
+                            let item = DashboardCardModel(id: card, cellViewModel: viewModel as? NSDictionary)
+
+                            snapshot.appendSections([section])
+                            snapshot.appendItems([item], toSection: section)
+                        }
+
+                    }
+
+                } else {
+                    let section = DashboardCardSection(id: card.rawValue)
+                    let item = DashboardCardModel(id: card)
+
+                    snapshot.appendSections([section])
+                    snapshot.appendItems([item], toSection: section)
+                }
+
+            }
+
+            completion(snapshot)
+        }, failure: { _ in
+
+        })
+    }
+}
+
+private extension BlogDashboardService {
+    /// Cards are a special case: they might not be a 1-1 relation
+    /// If the user has draft and scheduled posts, we show two cards
+    /// One for each. This function takes care of this
+    func parsePostCard(_ posts: NSDictionary) -> ([DashboardCardSection], [DashboardCardModel]) {
+        var sections: [DashboardCardSection] = []
+        var items: [DashboardCardModel] = []
+
+        let hasDrafts = (posts["draft"] as? Array<Any>)?.count ?? 0 > 0
+        let hasScheduled = (posts["scheduled"] as? Array<Any>)?.count ?? 0 > 0
```

Comment:
I found this statement a bit hard to read. What do you think of something like this?
```suggestion
        let draftsCount = (posts["draft"] as? Array<Any>)?.count ?? 0
        let scheduledCount = (posts["scheduled"] as? Array<Any>)?.count ?? 0

        let hasDrafts = draftsCount > 0
        let hasScheduled = scheduledCount > 0
```